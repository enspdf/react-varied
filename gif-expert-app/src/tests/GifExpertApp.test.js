import "@testing-library/jest-dom";

import { shallow } from "enzyme";
import React from "react";

import { GifExpertApp } from "../GifExpertApp";

describe("Test <GifExpertApp/>", () => {
  it("Should show the component", () => {
    const wrapper = shallow(<GifExpertApp />);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should show a categories list", () => {
    const categories = ["First", "Second"];
    const wrapper = shallow(<GifExpertApp defaultCategories={categories} />);
    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find("GifGrid").length).toBe(categories.length);
  });
});
