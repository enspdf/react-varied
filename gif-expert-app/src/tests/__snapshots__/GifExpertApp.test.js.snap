// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <GifExpertApp/> Should show a categories list 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="First"
      key="First"
    />
    <GifGrid
      category="Second"
      key="Second"
    />
  </ol>
</Fragment>
`;

exports[`Test <GifExpertApp/> Should show the component 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;
